extensions/amp-sidebar/1.0/sidebar.js

```diff
@@ -146,92 +159,118 @@ function SidebarWithRef(
   useLayoutEffect(() => {
     const sidebarElement = sidebarRef.current;
     const backdropElement = backdropRef.current;
-    if (!sidebarElement || !backdropElement) {
+    if (!sidebarElement || !backdropElement || !side) {
       return;
     }
 
-    if (!side) {
-      return;
-    }
-
-    let sidebarAnimation;
-    let backdropAnimation;
-    // "Make Visible" Animation
-    if (opened) {
-      const postVisibleAnim = () => {
-        safelySetStyles(sidebarElement, ANIMATION_STYLES_SIDEBAR_FINAL);
-        safelySetStyles(backdropElement, ANIMATION_STYLES_BACKDROP_FINAL);
-      };
-      if (!sidebarElement.animate || !backdropElement.animate) {
-        postVisibleAnim();
-        return;
+    const postVisibleAnim = () => {
+      safelySetStyles(sidebarElement, ANIMATION_STYLES_SIDEBAR_FINAL);
+      safelySetStyles(backdropElement, ANIMATION_STYLES_BACKDROP_FINAL);
+      setSidebarAnimation(null);
+      setBackdropAnimation(null);
+    };
+    const postInvisibleAnim = () => {
+      if (onAfterCloseRef.current) {
+        onAfterCloseRef.current();
       }
+      setSidebarAnimation(null);
+      setBackdropAnimation(null);
+      setMounted(false);
+    };
 
-      safelySetStyles(
-        sidebarElement,
-        side === Side.LEFT
-          ? ANIMATION_STYLES_SIDEBAR_LEFT_INIT
-          : ANIMATION_STYLES_SIDEBAR_RIGHT_INIT
-      );
-      safelySetStyles(backdropElement, ANIMATION_STYLES_BACKDROP_INIT);
-      sidebarAnimation = sidebarElement.animate(
-        side === Side.LEFT
-          ? ANIMATION_KEYFRAMES_SLIDE_IN_LEFT
-          : ANIMATION_KEYFRAMES_SLIDE_IN_RIGHT,
-        {
-          duration: ANIMATION_DURATION,
-          fill: 'both',
-          easing: ANIMATION_EASE_IN,
-        }
-      );
-      sidebarAnimation.onfinish = postVisibleAnim;
-      backdropAnimation = backdropElement.animate(ANIMATION_KEYFRAMES_FADE_IN, {
-        duration: ANIMATION_DURATION,
-        fill: 'both',
-        easing: ANIMATION_EASE_IN,
-      });
+    // currently animating
+    if (sidebarAnimation) {
```

Comment:
Could we still preserve the old structure but sort move the old cleanup function to the top? E.g.

```
const postVisibleAnim = () => {
  ...
  lastAnimationDirection.current = null;
  lastAnimation[i] = null;
};

if (lastAnimationOpen.current != null) {
  if (lastAnimationOpen.current === opened) {
    animations[0].onfinish = ...;
    animations[i].reverse();
  } else {
   // Is this even possible? If so, cancel or let it complete.
  }
  return;
}

if (opened) {
} else {
}
```

Reply:
Thanks, updated in the latest.